src/nccl_ofi_rdma.cpp

```diff
@@ -1967,20 +2017,29 @@ static int ofi_process_cq_rail(nccl_net_ofi_rdma_ep_t *ep, nccl_net_ofi_ep_rail_
 		/* Receive completions for the given endpoint */
 		rc = fi_cq_read(rail->cq, cqe_buffers, cq_read_count);
 		if (rc > 0) {
-			ret = process_completions(cqe_buffers, rc, rdma_endpoint_get_device(ep), rail->rail_id);
+			ret = rdma_process_completions(cqe_buffers, rc, rdma_endpoint_get_device(ep), rail->rail_id);
 			if (OFI_UNLIKELY(ret != 0))
 				goto exit;
 		} else if (OFI_UNLIKELY(rc == -FI_EAVAIL)) {
-			ret = process_err_completion(rdma_endpoint_get_device(ep), rail);
-			if (ret == 0) {
-				/* Error entry not available yet */
-				break;
-			} else if (ret == -FI_ECANCELED) {
-				/* Non-fatal cancellation event -- see comment in
-				   process_err_completion. Ignore. */
+
+			struct fi_cq_err_entry err_entry = {};
```

Comment:
I don't understand why you feel the need to zero this out before LIbfabric fills it in?